buildvertical.targets

```diff
@@ -0,0 +1,204 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <UsingTask TaskName="FindBestConfiguration" AssemblyFile="$(CoreFxToolsTaskDir)CoreFx.Tools.dll"/>
+
+  <Import Project="buildvertical.props" />
+
+  <PropertyGroup>
+    <ImportedBuildVerticalTargets>true</ImportedBuildVerticalTargets>
+  </PropertyGroup>
+  
+  <!-- Add vertical projects to the project group for BuildAllProjects -->
+  <Target Name="AddVerticalProjects"
+          Inputs="%(Project.Identity)"
+          Outputs="fake"
+          DependsOnTargets="DetermineProjectsConfiguration;TrimUnconvertedProjects"
+          BeforeTargets="AnnotateProjectsWithConfiguration">
+    <ItemGroup>
+      <OmitProject Include="%(Project.Identity)" Condition="'%(Project.OmitReason)' != ''">
+        <OmitReason>%(Project.OmitReason)</OmitReason>
+      </OmitProject>
+      <Project Remove="@(OmitProject)" />
+    </ItemGroup>
+  </Target>
+
+  <!-- The initial shaking target for trimming down applicable projects for specified vertical -->
+  <Target Name="DetermineProjectsConfiguration"
+          Inputs="%(Project.Identity)"
+          Outputs="fake"
+          Returns="@(Project)"
+          DependsOnTargets="TrimUnconvertedProjects"
+          BeforeTargets="AnnotateProjectsWithConfiguration">
+    <PropertyGroup>
+      <_VerticalProject>%(Project.Identity)</_VerticalProject>
+      <_VerticalProjectName>%(Project.FullPath)</_VerticalProjectName>
+      <_OmitReason>%(Project.OmitReason)</_OmitReason>
+    </PropertyGroup>
+
+    <!-- Clear the VerticalTargets item -->
+    <ItemGroup><VerticalTargets Remove="@(VerticalTargets)" /></ItemGroup>
+    <MSBuild Targets="FindBestConfiguration"
+             Projects="$(_VerticalProject)"
+             Condition="'$(_OmitReason)' == ''">
+      <Output TaskParameter="TargetOutputs"
+              ItemName="VerticalTargets" />
+    </MSBuild>
+
+    <PropertyGroup>
+      <_SelectedOSGroup>%(VerticalTargets.OSGroup)</_SelectedOSGroup>
+      <_SelectedTargetGroup>%(VerticalTargets.TargetGroup)</_SelectedTargetGroup>
+      <_Configuration>$(_SelectedTargetGroup)</_Configuration>
+      <_Configuration Condition="'$(_SelectedOSGroup)' != '' and '$(_SelectedOSGroup)' != 'AnyOS'">$(_Configuration)-$(_SelectedOSGroup)</_Configuration>
```

Comment:
Don't reconstruct, just use what you get from FindBestConfiguration.  It's already a configuration string.